pytestqt/wait_signal.py

```diff
@@ -54,11 +58,77 @@ def _quit_loop_by_timeout(self):
             self._loop.quit()
 
     def _cleanup(self):
+        # store timeout message before the data to construct it is lost
+        self._timeout_message = self.get_timeout_error_message()
         if self._timer is not None:
             _silent_disconnect(self._timer.timeout, self._quit_loop_by_timeout)
             self._timer.stop()
             self._timer = None
 
+    def get_timeout_error_message(self):
+        pass
+
+    def _extract_pyqt_signal_name(self, potential_pyqt_signal):
+        signal_name = potential_pyqt_signal.signal  # type: str
+        if type(signal_name) != str:
```

Comment:
I'm fine with replacing ``type()`` with ``isinstance()``. 
Regarding ``assert`` you're kind of disagreeing with yourself (see https://github.com/pytest-dev/pytest-qt/pull/141#discussion_r70802896 ), aren't you?